vertx-sql-client/src/main/java/io/vertx/sqlclient/impl/cache/PreparedStatementCacheManager.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (c) 2011-2020 Contributors to the Eclipse Foundation
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+ * which is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+ */
+
+package io.vertx.sqlclient.impl.cache;
+
+import io.vertx.core.AsyncResult;
+import io.vertx.core.Handler;
+import io.vertx.core.Promise;
+import io.vertx.sqlclient.impl.PreparedStatement;
+import io.vertx.sqlclient.impl.SocketConnectionBase;
+import io.vertx.sqlclient.impl.command.CloseStatementCommand;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Cache manager which manages the lifecycle of all cached prepared statements .
+ */
+public class PreparedStatementCacheManager {
```

Comment:
why does this needs to be renamed from ` PreparedStatementCache` to ` PreparedStatementCacheManager` ?

Reply:
It includes a `Map` recording inflight requests and manages the real cache which is used for caching `AsyncResult` of the prepare command responses. The `PreparedStatementResultLruCache` is a part of this and could also be pluggable in the future.

Reply:
can it be called cache like before and the map then lruMap ? I don't think we should over engineer this for now and see later in the future whether the needs arise in practice